### ISSUE TITLE
Question about DRP 

### ISSUE THREAD

**Issue by John33000** [wazuh/wazuh#4031]
Hi,

I would like know if I shutdown the server with elasticsearch, kibana and wazuh (like server had crashed). We restore backup of the VM  everything will work again ? Index of elasticsearch, wazuh ? 

Best regards,

**Comment by sergiovp** [comment]
Hello @John33000,

Firstly, Apologize us for the delay in our answer.

Regarding your question, I don't know what you exactly are asking for. 
Do you mean what happened if you restore a snapshot of your virtual machine?

In this case, I installed Wazuh-server, API, Elasticsearch and kibana. I registered two agents as you can see:
![image](https://user-images.githubusercontent.com/36703917/66906078-4be13b00-f007-11e9-877a-4788d6effaf1.png)

And I generated some alerts:
![image](https://user-images.githubusercontent.com/36703917/66906130-61eefb80-f007-11e9-99ad-1269c566008f.png)

Then, I did a snapshot of my VirtualBox machine and continued generating other alerts:
![image](https://user-images.githubusercontent.com/36703917/66906256-abd7e180-f007-11e9-8ff4-5c237e590c67.png)
Also, I removed my CentOS agent.

When I restarted my machine to the snapshot state and as you can see, I had exactly I had in the moment before I did the snapshot.
![image](https://user-images.githubusercontent.com/36703917/66906478-14bf5980-f008-11e9-90fa-9c031d238011.png)

Also, I had my CentOS agent:
![image](https://user-images.githubusercontent.com/36703917/66906546-33255500-f008-11e9-9152-e01a15f7ceb1.png)

To summarize, after restoring the backup, you will have everything exactly as you had before. 

The new alerts generated after the backup will be correctly indexed by Elasticsearch. On the other hand, you will lose everything you did after the snapshot(new registered agents, alerts... etc).

I hope I answered your question, otherwise, let me know. 
we'll be delighted to help you.

Best regards.

**Comment by John33000** [comment]
Hi,

Thank you for this answer. I have a second question, I have to use a Wazuh Cluster or Elasticsearch cluster from how many agents ? 

Best regards,

**Comment by sergiovp** [comment]
Hello @John33000, 

It depends on the quantity of EPS (Events Per Seconds) that your environment  produces. 
For example, a Windows agent produces more events (an event isn't the same as an alert, but each log message that is analyzed an may or may not trigger an alert) than a Linux agent. It also depends  on the customization of the agent configuration or if you use network devices that are sending syslog messages to the manager, if they are exposed to the Internet, etc.

If you give us some details about your use case, we will advise you better about the needed resources of your environment. 

Anyway, here are two example configurations:

**Case 1**: Monitoring 50-100 agents:

+ One node with Wazuh manager + ELK stack
+ 8 cores, 16GB RAM and a 128GB SSD hard-disk

**Case 2**: Monitoring 1000 agents

In this use case, my recommendation is to use Wazuh as cluster. Two nodes should be enough for 1000 agents depending on the events per second the each agent sends.
Assuming your Wazuh cluster has two nodes, they will be node01 and node02.

+ node01
    - Wazuh manager
    - Elasticsearch + Kibana + Logstash
    - 8GB RAM, 4 CPU, 128GB SSD disk.
    
+ node02
    - Wazuh manager
    - Filebeat
    - 4GB RAM, 2CPU, 64GB disk
+ Agents x 1000
    - Using a load balancer as manager IP
    
The agent hardware depends on how you will use them, other software will be used but the agent itself should not consume high resources, it's just a forwarder.

I hope it helps. As I said before, let me know your use case if you want a more customized recommendation.

Best regards.

**Comment by John33000** [comment]
Hi,
Thank you for your quality answer. 

With the last version of Elasticsearch and Wazuh, in your documentation it's written Logstash is not necessary. So I stop logstash service.

Currently we have 6 agent (4 Windows and 2 Linux) but we will increase between 100  and 500 agents. We will in case 1 or 2 ? 
And we have 4 vCPU, 8 Go of RAM 130Go of HDD (not SSD) and 1 VM with ELK, Kibana, Wazuh and Filebeat. 

How I can see the EPS ?

**Comment by sergiovp** [comment]
Hello John33000,

Effectively, it is not necessary to install Logstash. Filebeat is now in charge of log shipping from manager to elasticsearch so we don't longer need it.

In your case, you can choose either of the two configurations, but my recomendation is to use Wazuh and ELK as cluster.

In spite of what I told you, I would recommend the following configuration:

+ Node 1
	+ ELK + Kibana
	+ 8 cores and 32GB of RAM 

+ Node 2
	+ Wazuh Manager & Wazuh API
	+ 4 cores and 16 GB of RAM
	
In terms of the hard drive, SSD is highly recomended,  a HDD could be a bottle-neck.
	
However, in your case you may start with a Single-host architecture, then if you are having hard time to handle the load whether you can move on to distributed architecture or apply retention policy for the logs and add more memory.  But defiently to have high availability it is always recommended to have a distributed architecture ( Elasticsearch cluster + Wazuh cluster).

You can see the total EPS per 5 seconds in the file located at `/var/ossec/var/run/ossec-analysisd.state`. (Total events decoded).
As an approach, with the default configuration, between 1-2 EPS are generated by each agent.
Also, you can check our related documentation https://documentation.wazuh.com/3.10/user-manual/reference/statistics-files/ossec-analysisd-state.html

I hope I helped you. Let me know if you have another type of question.

Best regards.

**Comment by John33000** [comment]
Hi, 

Thank you for your help.

Best regards

**Comment by John33000** [comment]
Hi @sergiovp 

I open this case. 

What's difference between Wazuh Cluster and ELK Cluster ? In your case 2 you talk about Wazuh as cluster.
In documentation of Wazuh it's written Cluster of ELK not Cluster of Wazuh.
In the end we are going to have more than1000 Agents. It's better to have : 

Case 1 :

- Nodes 1 

1. - Wazuh manager
2. - Filebeat

- Nodes 2 : 

1. Elasticsearch
2. Kibana

- Nodes 3: 

1. Elasticsearch


Or 

Case 2 : 

- Nodes 1 :
1. Wazuh Manager
2. Elasticsearch
3. Kibana

- Nodes 2 : 

1. Wazuh Manager
2. Filebeat

Best Regards

**Comment by sergiovp** [comment]
Hello again @John33000,

The fundamental difference between Wazuh Cluster and ELK Cluster is as follows:

+ ELK Cluster: several physical or virtual nodes (two or more), placing its resources available to ELK.

+ Wazuh Cluster: several physical or virtual nodes (two or more), placing its resources available to a master Wazuh manager and one or more Wazuh manager workers.

As far as your case, I would recommend your first case with a minor  change, one more node of Elasticsearch. It will prevent "split brain".

You also have the full high availability option:

+ Wazuh cluster (Behind a Load balancer)
    - Node 1: Wazuh Master + Filebeat
    - Node 2:  Wazuh Worker + Filebeat
- Elasticsearch cluster (3 nodes)
    - Node 3: Elasticsearch + Kibana
    - Node 4: Elasticsearch
    - Node 5: Elasticsearch
    
I hope I helped you, in another case, let me know. I reopened the issue.

Best regards.

**Comment by John33000** [comment]
Hi @sergiovp,

Thanks for your anwser.
So for more than 1000 agents I will need to Nodes 4 with Elasticsearch ? 
And for hardware requirement : 

Case 1 :

- Nodes 1

   - Wazuh manager
   - Filebeat
4GB RAM, 2vCpu and 64GB HDD 

- Nodes 2 :

   - Elasticsearch
   - Kibana
8GB RAM, 4VCPU, 128GB SSD
- Nodes 3 :
   
   - Elasticsearch
8GB RAM, 4VCPU, 128GB SSD

- Nodes 4 : 

   - Elasticsearch
8GB RAM, 4VCPU, 128GB SSD

It's good ? It's expensive and big architecture.  No ? Nodes 4 it's necessary ?

**Comment by Zenidd** [comment]
Hello @John33000,

As @sergiovp  said above, the configuration really depends on how much EPS your endpoints produce, if you will monitor other types of devices (network devices), how long you will need the data to be online, etc..


For 1000-1500 agents, I think that the further configuration can fit:

- _2x_ **Wazuh manager and Filebeat node**: 4 cores, 16 GB of RAM and 1TB disk space. _Per node_.
- _1x or 2x_ **ElasticSearch and Kibana node**: 8 cores, 32 GB of RAM minimum and 64 GB max, 1 TB of disk space minimum. (However, this will depend on the data you store in Elastic). _Per node_.

Depending on the number of agents in your environment, you could add more Elastic&Kibana nodes and managers to your architecture with these hardware requirements for scalability.

I hope it helps. Let me know if you have any other questions. Greetings, 
JP Sáez

**Comment by John33000** [comment]
Hi @Zenidd,

Thank you for your help.
I will follow the recommandation of @sergiovp "As far as your case, I would recommend your first case with a minor change, one more node of Elasticsearch. It will prevent "split brain"."
I will follow my configuration. Currently we have 42 EPS for 7 Agents. Eventually we can have 700 Agents and aproximately 4200 EPS. 

Best Regards.

**Comment by Zenidd** [comment]
That's perfect @John33000, count on us if you need further guidance with the hardware sizing guidelines. On the other hand, for further questions please we would like you to use our [#community Slack channel](https://wazuh.com/community/join-us-on-slack/) or the [Wazuh mail list](https://groups.google.com/forum/?hl=es#!forum/wazuh).

Regards, JP Sáez
